<Type Name="DirectoryNotFoundException" FullName="System.IO.DirectoryNotFoundException">
  <Metadata>
    <Meta Name="ms.openlocfilehash" Value="465bf8711c842543ef0592f4baebbfbfdcc8fa37" />
    <Meta Name="ms.sourcegitcommit" Value="5a49536d99d2d0b54e4cb7280870903e043272df" />
    <Meta Name="ms.translationtype" Value="HT" />
    <Meta Name="ms.contentlocale" Value="it-IT" />
    <Meta Name="ms.lasthandoff" Value="07/03/2018" />
    <Meta Name="ms.locfileid" Value="37652474" />
  </Metadata>
  <TypeSignature Language="C#" Value="public class DirectoryNotFoundException : System.IO.IOException" />
  <TypeSignature Language="ILAsm" Value=".class public auto ansi serializable beforefieldinit DirectoryNotFoundException extends System.IO.IOException" />
  <TypeSignature Language="DocId" Value="T:System.IO.DirectoryNotFoundException" />
  <TypeSignature Language="VB.NET" Value="Public Class DirectoryNotFoundException&#xA;Inherits IOException" />
  <TypeSignature Language="C++ CLI" Value="public ref class DirectoryNotFoundException : System::IO::IOException" />
  <TypeSignature Language="F#" Value="type DirectoryNotFoundException = class&#xA;    inherit IOException" />
  <AssemblyInfo>
    <AssemblyName>System.Runtime</AssemblyName>
    <AssemblyVersion>4.0.20.0</AssemblyVersion>
    <AssemblyVersion>4.1.0.0</AssemblyVersion>
    <AssemblyVersion>4.2.0.0</AssemblyVersion>
    <AssemblyVersion>4.2.1.0</AssemblyVersion>
  </AssemblyInfo>
  <AssemblyInfo>
    <AssemblyName>mscorlib</AssemblyName>
    <AssemblyVersion>1.0.5000.0</AssemblyVersion>
    <AssemblyVersion>2.0.0.0</AssemblyVersion>
    <AssemblyVersion>2.0.5.0</AssemblyVersion>
    <AssemblyVersion>4.0.0.0</AssemblyVersion>
  </AssemblyInfo>
  <AssemblyInfo>
    <AssemblyName>netstandard</AssemblyName>
    <AssemblyVersion>2.0.0.0</AssemblyVersion>
  </AssemblyInfo>
  <Base>
    <BaseTypeName>System.IO.IOException</BaseTypeName>
  </Base>
  <Interfaces />
  <Attributes>
    <Attribute>
      <AttributeName>System.Runtime.InteropServices.ComVisible(true)</AttributeName>
    </Attribute>
  </Attributes>
  <Docs>
    <summary>
      <span data-ttu-id="47967-101">Eccezione che viene generata quando non viene trovata parte di un file o di una directory.</span>
      <span class="sxs-lookup">
        <span data-stu-id="47967-101">The exception that is thrown when part of a file or directory cannot be found.</span>
      </span>
    </summary>
    <remarks>
      <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="47967-102">`DirectoryNotFoundException` utilizza COR_E_DIRECTORYNOTFOUND HRESULT che ha il valore HRESULT:0x80070003.</span><span class="sxs-lookup"><span data-stu-id="47967-102">`DirectoryNotFoundException` uses the HRESULT COR_E_DIRECTORYNOTFOUND which has the value 0x80070003.</span></span> <span data-ttu-id="47967-103">Si noti che `DirectoryNotFoundException` viene generata anche quando i programmi di interoperabilità COM vedere STG_E_PATHNOTFOUND l'HRESULT, che ha il valore 0x80030003.</span><span class="sxs-lookup"><span data-stu-id="47967-103">Note that `DirectoryNotFoundException` is also thrown when COM interop programs see the HRESULT STG_E_PATHNOTFOUND, which has the value 0x80030003.</span></span>  
  
 <span data-ttu-id="47967-104">Se il codice non dispone di <xref:System.Security.Permissions.FileIOPermissionAttribute.PathDiscovery%2A> autorizzazione, il messaggio di errore per questa eccezione può contenere solo i nomi di file o directory invece percorsi completi.</span><span class="sxs-lookup"><span data-stu-id="47967-104">If your code does not have <xref:System.Security.Permissions.FileIOPermissionAttribute.PathDiscovery%2A> permission, the error message for this exception may only contain file or directory names instead of fully qualified paths.</span></span>  
  
   
  
## Examples  
 <span data-ttu-id="47967-105">Nell'esempio seguente viene illustrato come forzare e ripristinare da un <xref:System.IO.DirectoryNotFoundException>.</span><span class="sxs-lookup"><span data-stu-id="47967-105">The following example shows how to force and recover from a <xref:System.IO.DirectoryNotFoundException>.</span></span>  
  
 [!code-csharp[DirectoryNotFoundException#1](~/samples/snippets/csharp/VS_Snippets_CLR/DirectoryNotFoundException/cs/program.cs#1)]
 [!code-vb[DirectoryNotFoundException#1](~/samples/snippets/visualbasic/VS_Snippets_CLR/DirectoryNotFoundException/VB/DirectoryNotFoundexception.vb#1)]  
  
 ]]></format>
    </remarks>
    <altmember cref="T:System.Exception" />
  </Docs>
  <Members>
    <MemberGroup MemberName=".ctor">
      <AssemblyInfo>
        <AssemblyName>System.Runtime</AssemblyName>
        <AssemblyVersion>4.0.20.0</AssemblyVersion>
        <AssemblyVersion>4.1.0.0</AssemblyVersion>
        <AssemblyVersion>4.2.0.0</AssemblyVersion>
      </AssemblyInfo>
      <Docs>
        <summary>
          <span data-ttu-id="47967-106">Inizializza una nuova istanza della classe <see cref="T:System.IO.DirectoryNotFoundException" /> con la relativa stringa di messaggio.</span>
          <span class="sxs-lookup">
            <span data-stu-id="47967-106">Initializes a new instance of the <see cref="T:System.IO.DirectoryNotFoundException" /> class with its message string.</span>
          </span>
        </summary>
      </Docs>
    </MemberGroup>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="public DirectoryNotFoundException ();" />
      <MemberSignature Language="ILAsm" Value=".method public hidebysig specialname rtspecialname instance void .ctor() cil managed" />
      <MemberSignature Language="DocId" Value="M:System.IO.DirectoryNotFoundException.#ctor" />
      <MemberSignature Language="VB.NET" Value="Public Sub New ()" />
      <MemberSignature Language="C++ CLI" Value="public:&#xA; DirectoryNotFoundException();" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>System.Runtime</AssemblyName>
        <AssemblyVersion>4.0.20.0</AssemblyVersion>
        <AssemblyVersion>4.1.0.0</AssemblyVersion>
        <AssemblyVersion>4.2.0.0</AssemblyVersion>
        <AssemblyVersion>4.2.1.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>mscorlib</AssemblyName>
        <AssemblyVersion>1.0.5000.0</AssemblyVersion>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
        <AssemblyVersion>2.0.5.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>netstandard</AssemblyName>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <Parameters />
      <Docs>
        <summary>
          <span data-ttu-id="47967-107">Inizializza una nuova istanza della classe <see cref="T:System.IO.DirectoryNotFoundException" /> con la relativa stringa di messaggio impostata su un messaggio fornito dal sistema e il relativo HRESULT impostato su COR_E_DIRECTORYNOTFOUND.</span>
          <span class="sxs-lookup">
            <span data-stu-id="47967-107">Initializes a new instance of the <see cref="T:System.IO.DirectoryNotFoundException" /> class with its message string set to a system-supplied message and its HRESULT set to COR_E_DIRECTORYNOTFOUND.</span>
          </span>
        </summary>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="47967-108">Questo costruttore inizializza il <xref:System.Exception.Message%2A?displayProperty=nameWithType> proprietà della nuova istanza di un messaggio fornito dal sistema che descrive l'errore, ad esempio "Impossibile trovare la directory specificata."</span><span class="sxs-lookup"><span data-stu-id="47967-108">This constructor initializes the <xref:System.Exception.Message%2A?displayProperty=nameWithType> property of the new instance to a system-supplied message that describes the error, such as "Could not find the specified directory."</span></span> <span data-ttu-id="47967-109">Questo messaggio prende in considerazione le impostazioni cultura correnti del sistema.</span><span class="sxs-lookup"><span data-stu-id="47967-109">This message takes into account the current system culture.</span></span>  
  
 <span data-ttu-id="47967-110">Il <xref:System.Exception.InnerException%2A?displayProperty=nameWithType> proprietà della nuova istanza viene inizializzata su `null`.</span><span class="sxs-lookup"><span data-stu-id="47967-110">The <xref:System.Exception.InnerException%2A?displayProperty=nameWithType> property of the new instance is initialized to `null`.</span></span>  
  
   
  
## Examples  
 <span data-ttu-id="47967-111">Esempio di codice seguente viene illustrato come chiamare il <xref:System.IO.DirectoryNotFoundException.%23ctor%2A> costruttore.</span><span class="sxs-lookup"><span data-stu-id="47967-111">The following code example demonstrates how to call the <xref:System.IO.DirectoryNotFoundException.%23ctor%2A> constructor.</span></span>  
  
 [!code-csharp[DirectoryNotFoundException#1](~/samples/snippets/csharp/VS_Snippets_CLR/DirectoryNotFoundException/cs/program.cs#1)]
 [!code-vb[DirectoryNotFoundException#1](~/samples/snippets/visualbasic/VS_Snippets_CLR/DirectoryNotFoundException/VB/DirectoryNotFoundexception.vb#1)]  
  
 ]]></format>
        </remarks>
      </Docs>
    </Member>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="public DirectoryNotFoundException (string message);" />
      <MemberSignature Language="ILAsm" Value=".method public hidebysig specialname rtspecialname instance void .ctor(string message) cil managed" />
      <MemberSignature Language="DocId" Value="M:System.IO.DirectoryNotFoundException.#ctor(System.String)" />
      <MemberSignature Language="VB.NET" Value="Public Sub New (message As String)" />
      <MemberSignature Language="C++ CLI" Value="public:&#xA; DirectoryNotFoundException(System::String ^ message);" />
      <MemberSignature Language="F#" Value="new System.IO.DirectoryNotFoundException : string -&gt; System.IO.DirectoryNotFoundException" Usage="new System.IO.DirectoryNotFoundException message" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>System.Runtime</AssemblyName>
        <AssemblyVersion>4.0.20.0</AssemblyVersion>
        <AssemblyVersion>4.1.0.0</AssemblyVersion>
        <AssemblyVersion>4.2.0.0</AssemblyVersion>
        <AssemblyVersion>4.2.1.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>mscorlib</AssemblyName>
        <AssemblyVersion>1.0.5000.0</AssemblyVersion>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
        <AssemblyVersion>2.0.5.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>netstandard</AssemblyName>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <Parameters>
        <Parameter Name="message" Type="System.String" />
      </Parameters>
      <Docs>
        <param name="message">
          <span data-ttu-id="47967-112">
            <see cref="T:System.String" /> che descrive l'errore.</span>
          <span class="sxs-lookup">
            <span data-stu-id="47967-112">A <see cref="T:System.String" /> that describes the error.</span>
          </span>
          <span data-ttu-id="47967-113">Il contenuto del <c>messaggio</c> deve essere facilmente comprensibile.</span>
          <span class="sxs-lookup">
            <span data-stu-id="47967-113">The content of <c>message</c> is intended to be understood by humans.</span>
          </span>
          <span data-ttu-id="47967-114">Il chiamante del costruttore deve garantire che la stringa sia stata localizzata in base alle impostazioni cultura correnti del sistema.</span>
          <span class="sxs-lookup">
            <span data-stu-id="47967-114">The caller of this constructor is required to ensure that this string has been localized for the current system culture.</span>
          </span>
        </param>
        <summary>
          <span data-ttu-id="47967-115">Inizializza una nuova istanza della classe <see cref="T:System.IO.DirectoryNotFoundException" /> con la relativa stringa di messaggio impostata su <paramref name="message" /> e il relativo HRESULT impostato su COR_E_DIRECTORYNOTFOUND.</span>
          <span class="sxs-lookup">
            <span data-stu-id="47967-115">Initializes a new instance of the <see cref="T:System.IO.DirectoryNotFoundException" /> class with its message string set to <paramref name="message" /> and its HRESULT set to COR_E_DIRECTORYNOTFOUND.</span>
          </span>
        </summary>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="47967-116">Questo costruttore inizializza la <xref:System.Exception.Message%2A?displayProperty=nameWithType> proprietà della nuova istanza utilizzando parametro `message`.</span><span class="sxs-lookup"><span data-stu-id="47967-116">This constructor initializes the <xref:System.Exception.Message%2A?displayProperty=nameWithType> property of the new instance using `message`.</span></span>  
  
 <span data-ttu-id="47967-117">Il <xref:System.Exception.InnerException%2A?displayProperty=nameWithType> proprietà della nuova istanza viene inizializzata su `null`.</span><span class="sxs-lookup"><span data-stu-id="47967-117">The <xref:System.Exception.InnerException%2A?displayProperty=nameWithType> property of the new instance is initialized to `null`.</span></span>  
  
 ]]></format>
        </remarks>
      </Docs>
    </Member>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="protected DirectoryNotFoundException (System.Runtime.Serialization.SerializationInfo info, System.Runtime.Serialization.StreamingContext context);" />
      <MemberSignature Language="ILAsm" Value=".method familyhidebysig specialname rtspecialname instance void .ctor(class System.Runtime.Serialization.SerializationInfo info, valuetype System.Runtime.Serialization.StreamingContext context) cil managed" />
      <MemberSignature Language="DocId" Value="M:System.IO.DirectoryNotFoundException.#ctor(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)" />
      <MemberSignature Language="VB.NET" Value="Protected Sub New (info As SerializationInfo, context As StreamingContext)" />
      <MemberSignature Language="C++ CLI" Value="protected:&#xA; DirectoryNotFoundException(System::Runtime::Serialization::SerializationInfo ^ info, System::Runtime::Serialization::StreamingContext context);" />
      <MemberSignature Language="F#" Value="new System.IO.DirectoryNotFoundException : System.Runtime.Serialization.SerializationInfo * System.Runtime.Serialization.StreamingContext -&gt; System.IO.DirectoryNotFoundException" Usage="new System.IO.DirectoryNotFoundException (info, context)" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>mscorlib</AssemblyName>
        <AssemblyVersion>1.0.5000.0</AssemblyVersion>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
        <AssemblyVersion>2.0.5.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>netstandard</AssemblyName>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>System.Runtime</AssemblyName>
        <AssemblyVersion>4.2.0.0</AssemblyVersion>
        <AssemblyVersion>4.2.1.0</AssemblyVersion>
      </AssemblyInfo>
      <Parameters>
        <Parameter Name="info" Type="System.Runtime.Serialization.SerializationInfo" />
        <Parameter Name="context" Type="System.Runtime.Serialization.StreamingContext" />
      </Parameters>
      <Docs>
        <param name="info">
          <span data-ttu-id="47967-118">Oggetto <see cref="T:System.Runtime.Serialization.SerializationInfo" /> che contiene i dati oggetto serializzati relativi all'eccezione generata.</span>
          <span class="sxs-lookup">
            <span data-stu-id="47967-118">The <see cref="T:System.Runtime.Serialization.SerializationInfo" /> that holds the serialized object data about the exception being thrown.</span>
          </span>
        </param>
        <param name="context">
          <span data-ttu-id="47967-119">Oggetto <see cref="T:System.Runtime.Serialization.StreamingContext" /> contenente le informazioni contestuali sull'origine o sulla destinazione.</span>
          <span class="sxs-lookup">
            <span data-stu-id="47967-119">The <see cref="T:System.Runtime.Serialization.StreamingContext" /> that contains contextual information about the source or destination.</span>
          </span>
        </param>
        <summary>
          <span data-ttu-id="47967-120">Inizializza una nuova istanza della classe <see cref="T:System.IO.DirectoryNotFoundException" /> con le informazioni di serializzazione e di contesto specificate.</span>
          <span class="sxs-lookup">
            <span data-stu-id="47967-120">Initializes a new instance of the <see cref="T:System.IO.DirectoryNotFoundException" /> class with the specified serialization and context information.</span>
          </span>
        </summary>
        <remarks>To be added.</remarks>
      </Docs>
    </Member>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="public DirectoryNotFoundException (string message, Exception innerException);" />
      <MemberSignature Language="ILAsm" Value=".method public hidebysig specialname rtspecialname instance void .ctor(string message, class System.Exception innerException) cil managed" />
      <MemberSignature Language="DocId" Value="M:System.IO.DirectoryNotFoundException.#ctor(System.String,System.Exception)" />
      <MemberSignature Language="VB.NET" Value="Public Sub New (message As String, innerException As Exception)" />
      <MemberSignature Language="C++ CLI" Value="public:&#xA; DirectoryNotFoundException(System::String ^ message, Exception ^ innerException);" />
      <MemberSignature Language="F#" Value="new System.IO.DirectoryNotFoundException : string * Exception -&gt; System.IO.DirectoryNotFoundException" Usage="new System.IO.DirectoryNotFoundException (message, innerException)" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>System.Runtime</AssemblyName>
        <AssemblyVersion>4.0.20.0</AssemblyVersion>
        <AssemblyVersion>4.1.0.0</AssemblyVersion>
        <AssemblyVersion>4.2.0.0</AssemblyVersion>
        <AssemblyVersion>4.2.1.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>mscorlib</AssemblyName>
        <AssemblyVersion>1.0.5000.0</AssemblyVersion>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
        <AssemblyVersion>2.0.5.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>netstandard</AssemblyName>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <Parameters>
        <Parameter Name="message" Type="System.String" />
        <Parameter Name="innerException" Type="System.Exception" />
      </Parameters>
      <Docs>
        <param name="message">
          <span data-ttu-id="47967-121">Messaggio di errore in cui viene indicato il motivo dell'eccezione.</span>
          <span class="sxs-lookup">
            <span data-stu-id="47967-121">The error message that explains the reason for the exception.</span>
          </span>
        </param>
        <param name="innerException">
          <span data-ttu-id="47967-122">Eccezione che ha determinato l'eccezione corrente.</span>
          <span class="sxs-lookup">
            <span data-stu-id="47967-122">The exception that is the cause of the current exception.</span>
          </span>
          <span data-ttu-id="47967-123">Se il parametro <c>innerException</c> non è <see langword="null" />, l'eccezione corrente viene generata in un blocco <see langword="catch" /> in cui viene gestita l'eccezione interna.</span>
          <span class="sxs-lookup">
            <span data-stu-id="47967-123">If the <c>innerException</c> parameter is not <see langword="null" />, the current exception is raised in a <see langword="catch" /> block that handles the inner exception.</span>
          </span>
        </param>
        <summary>
          <span data-ttu-id="47967-124">Inizializza una nuova istanza della classe <see cref="T:System.IO.DirectoryNotFoundException" /> con un messaggio di errore specificato e un riferimento all'eccezione interna che è la causa dell'eccezione corrente.</span>
          <span class="sxs-lookup">
            <span data-stu-id="47967-124">Initializes a new instance of the <see cref="T:System.IO.DirectoryNotFoundException" /> class with a specified error message and a reference to the inner exception that is the cause of this exception.</span>
          </span>
        </summary>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="47967-125">Un'eccezione generata come risultato diretto di un'eccezione precedente deve includere un riferimento all'eccezione precedente nella proprietà <xref:System.Exception.InnerException%2A>.</span><span class="sxs-lookup"><span data-stu-id="47967-125">An exception that is thrown as a direct result of a previous exception should include a reference to the previous exception in the <xref:System.Exception.InnerException%2A> property.</span></span> <span data-ttu-id="47967-126">La proprietà <xref:System.Exception.InnerException%2A> restituisce lo stesso valore passato al costruttore oppure `null` nel caso in cui la proprietà <xref:System.Exception.InnerException%2A> non fornisca il valore di eccezione interna al costruttore.</span><span class="sxs-lookup"><span data-stu-id="47967-126">The <xref:System.Exception.InnerException%2A> property returns the same value that is passed into the constructor, or `null` if the <xref:System.Exception.InnerException%2A> property does not supply the inner exception value to the constructor.</span></span>  
  
 <span data-ttu-id="47967-127">Nella tabella seguente vengono illustrati i valori di proprietà iniziali per un'istanza di <xref:System.IO.DirectoryNotFoundException>.</span><span class="sxs-lookup"><span data-stu-id="47967-127">The following table shows the initial property values for an instance of <xref:System.IO.DirectoryNotFoundException>.</span></span>  
  
|<span data-ttu-id="47967-128">Proprietà</span><span class="sxs-lookup"><span data-stu-id="47967-128">Property</span></span>|<span data-ttu-id="47967-129">Valore</span><span class="sxs-lookup"><span data-stu-id="47967-129">Value</span></span>|  
|--------------|-----------|  
|<xref:System.Exception.InnerException%2A>|<span data-ttu-id="47967-130">Riferimento all'eccezione interna.</span><span class="sxs-lookup"><span data-stu-id="47967-130">The inner exception reference.</span></span>|  
|<xref:System.Exception.Message%2A>|<span data-ttu-id="47967-131">Stringa del messaggio di errore.</span><span class="sxs-lookup"><span data-stu-id="47967-131">The error message string.</span></span>|  
  
 ]]></format>
        </remarks>
        <altmember cref="T:System.Exception" />
      </Docs>
    </Member>
  </Members>
</Type>